nens_auth_client/views.py

```diff
@@ -197,12 +192,38 @@ def logout(request):
 
     # Redirect to authorization server
     logout_uri = request.build_absolute_uri(
-        reverse(settings.NENS_AUTH_URL_NAMESPACE + "logout")
+        reverse(settings.NENS_AUTH_URL_NAMESPACE + "logout-success")
     )
     client = get_oauth_client()
     return client.logout_redirect(request, logout_uri)
 
 
+@cache_control(no_store=True)
+def logout_success(request):
+    """Callback url for logout.
+
+    See the README for a description of the logout flow.
+
+    Query parameters:
+      next: the URL to redirect to after logout. If absolute, it
+        must match the domain of this request. Optional, default is set by
+        settings.NENS_AUTH_DEFAULT_LOGOUT_URL
+
+    Response:
+    - if user is logged out locally: HTTP 302 Redirect to the 'next' query parameter
+      that is stored in the session (or default logout url if unavailable).
+    - if user is logged in locally: HTTP 302 Redirect to the logout view
+    """
+    # If a user is still authenticated: redirect to logout view.
+    if request.user.is_authenticated:
+        raise PermissionDenied("Logout failure")
```

Comment:
Een "permission denied" raisen, dat redirect je normaliter toch naar de inlogpagina i.p.v. de logout pagina?

Reply:
Oh die annotation klopte niet meer. Eerst deed ik een redirect naar "logout" maar toen realiseerde ik me dat dit ongewenst gebruik is, dus dat we beter een error kunnen geven.